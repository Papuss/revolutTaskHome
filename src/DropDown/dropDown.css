@import '~antd/dist/antd.css';

.dropdown-component {    
    display: inline-flex;
    padding: 5px;
    border-radius: 10px;
    background-color: lightgray;
}

.dropdown-component .options {
    border: none;
    outline: none;
    background-color: lightgray;
    margin-left: 10px;
}

.dropdown-component .options select {  
    border: none;
    outline: none;
}