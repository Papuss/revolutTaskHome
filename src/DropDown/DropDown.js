import "./dropDown.css";
import {Select} from "antd";
import React from 'react';

const { Option } = Select;

function renderCurrencies (props) {
    return (
        props.rates &&
        Object.keys(props.rates).map((currency, index) => (
            <Option value={currency} key={index}>{currency}</Option>
        ))
    );
}

function DropDown (props) {
    return (
        <div className="dropdown-component" id={props.id}>
        <div className="options">
            <Select defaultValue={props.currency} style={{ width: 120 }}
            onChange={newCurr => {
                props.onChange(newCurr);                                      
            }}>
                {renderCurrencies(props)}
            </Select>
        </div>
    </div>
    )
}

export default DropDown;