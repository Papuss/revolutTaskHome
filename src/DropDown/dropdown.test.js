import DropDown from './DropDown';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow } from 'enzyme';
import React from 'react';

Enzyme.configure({ adapter: new Adapter() });

describe('Renders Dropdown component', () => {
it('renders as expected', () => {
    const wrapper = shallow(
        <DropDown 
            id="fromCurrency"
            rates={{ "GBP": "2650", "USD": "1200", "EUR": "8500" }}
            currency={"HUF"}
            onChange={() => {}}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
});