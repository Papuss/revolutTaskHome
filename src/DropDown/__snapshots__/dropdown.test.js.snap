// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Dropdown component renders as expected 1`] = `ShallowWrapper {}`;
